src/components/VmDetails/cards/DetailsCard/index.js

```diff
@@ -253,6 +278,13 @@ class DetailsCard extends React.Component {
     if (prevProps.templates !== this.props.templates) {
       this.setState({ templateList: createTemplateList(this.props.templates) }) // eslint-disable-line react/no-did-update-set-state
     }
+
+    if (prevProps.operatingSystems !== this.props.operatingSystems) {
```

Comment:
Is this enough of a check for when the `osList` needs to be rebuilt?  Since `osList` is dependent on the VM (to get the cluster) and Clusters (to get the architecture), may be worth adding a check for new props on those as well.